src/Helper/BeskedfordelerHelper.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Drupal\os2forms_digital_post\Helper;
+
+use DigitalPost\MeMo\Message as MeMoMessage;
+use Drupal\Core\Database\Connection;
+use Drupal\os2forms_digital_post\Exception\InvalidMessage;
+use Drupal\os2forms_digital_post\Model\Message;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Beskedfordeler helper.
+ */
+class BeskedfordelerHelper {
+  use LoggerAwareTrait;
+
+  private const TABLE_NAME = 'os2forms_digital_post_beskedfordeler';
+
+  /**
+   * The database.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  private Connection $database;
+
+  /**
+   * The MeMo helper.
+   *
+   * @var MeMoHelper
+   */
+  private MeMoHelper $meMoHelper;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(Connection $database, MeMoHelper $meMoHelper, LoggerInterface $logger) {
+    $this->database = $database;
+    $this->meMoHelper = $meMoHelper;
+    $this->setLogger($logger);
+  }
+
+  /**
+   * Save MeMo message in database.
+   */
+  public function createMessage(int $submissionId, MeMoMessage $message, string $receipt) {
+    $messageUUID = $message->getMessageHeader()->getMessageUUID();
+    $message = $this->meMoHelper->message2dom($message)->saveXML();
+
+    return $this->database
+      ->insert(self::TABLE_NAME)
+      ->fields([
+        'submission_id' => $submissionId,
+        'message_uuid' => $messageUUID,
+        'message' => $message,
+        'receipt' => $receipt,
+      ])
+      ->execute();
+  }
+
+  /**
+   * Load message.
+   *
+   * @param string $messageUUID
+   *   The message UUID.
+   *
+   * @return \Drupal\os2forms_digital_post\Model\Message|null
+   *   The message.
+   */
+  public function loadMessage(string $messageUUID): ?Message {
+    $messages = $this->database
+      ->select(self::TABLE_NAME, 'm')
+      ->fields('m')
+      ->condition('message_uuid', $messageUUID)
+      ->execute()
+      ->fetchAll(\PDO::FETCH_CLASS, Message::class);
+
+    return reset($messages) ?: NULL;
```

Comment:
Why `reset()`? Could there ever be more than one?

Reply:
No, but `$messages` is a list of messages and I want just a single one.